stingray/crosscorrelation.py

```diff
@@ -74,23 +81,69 @@ def __init__(self, lc1=None, lc2=None, mode='same'):
         self.mode = mode.lower()
         self.lc1 = None
         self.lc2 = None
+        self.cross = None
 
         # Populate all attributes by ``None` if user passes no lightcurve data
         if lc1 is None or lc2 is None:
             if lc1 is not None or lc2 is not None:
                 raise TypeError("You can't do a cross correlation with just one "
                                 "light curve!")
+
             else:
-                # both lc1 and lc2 are ``Non.
-                self.corr = None
-                self.time_shift = None
-                self.time_lags = None
-                self.dt = None
-                self.n = None
-                return
+                if cross is None:
+                    # all object input params are ``None``
+                    self.corr = None
+                    self.time_shift = None
+                    self.time_lags = None
+                    self.dt = None
+                    self.n = None
+                else:
+                    self._make_cross_corr(cross)
+                    return
         else:
             self._make_corr(lc1, lc2)
 
+    def _make_cross_corr(self, cross):
+
+        """
+        Do some checks on the cross spectrum supplied to the method,
+        and then calculate the time shifts, time lags and cross correlation.
+
+        Parameters
+        ----------
+        cross: :class:`stingray.Crossspectrum` object
+            The crossspectrum, averaged or not.
+
+        """
+
+        if not isinstance(cross, crossspectrum.Crossspectrum):
+            if not isinstance(cross, crossspectrum.AveragedCrossspectrum):
+                raise TypeError("cross must be a crossspectrum.Crossspectrum \
+                        or crossspectrum.AveragedCrossspectrum object")
+
+        if self.cross is None:
+            self.cross = cross
+            self.dt = 1/(cross.df * cross.n)
+        if self.dt is None:
+            self.dt = 1/(cross.df * cross.n)
+
+
+        prelim_corr = abs(ifft(cross.power).real) #keep only the real
+        self.n = len(prelim_corr)
+
+        # ifft spits out an array that looks like [0,1,...n,-n,...-1]
+        # where n is the last positive frequency
+        # correcting for this by putting them in order
+
+        times = np.fft.fftfreq(self.n, cross.df)
+        time, corr = (list(t) for t in zip(*sorted(zip(times, prelim_corr))))
```

Comment:
What does this line do?

Reply:
Here's an alternative way of writing this: `time, corr = np.array(sorted(zip(times, prelim_corr))).T`